pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -122,6 +122,12 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration, featur
 	if kc.RegistryPullQPS < 0 {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: registryPullQPS (--registry-qps) %v must not be a negative number", kc.RegistryPullQPS))
 	}
+	if kc.MaxParallelImagePulls != nil && *kc.MaxParallelImagePulls < 2 {
```

Comment:
Why we need (>=2) constraint here? Shouldn't MaxParallelImagePulls == 1 is a valid input, and effectively change the behavior equal to serializing pulling? In that case we can eventually deprecate both parallelImagePulls and serializeImagePulls, and leave only maxParallelImagePulls?

Reply:
good point. I updated the code to
1. allow `MaxParallelImagePulls==1`
2. allow `SerializeImagePulls=true && MaxParallelImagePulls==1`, where a serial image puller will be used